ankidroid/Anki-Android
# Sync from reviewer when we didn't want through the deck picker

Sync enabled in the reviewer when the deck picker was not used

Widgets and shortcuts allows to directly review a deck.
This means that automated sync, or even manual sync, is never
available. Except if the user close the reviewer and then manually
open ankidroid. This create a lot of friction on a very important
feature.

This commit enable the sync feature in the reviewer, but hides it
unless the intent actually requested it.

In my opinion, it should be always available. It's available in anki
upstream from the reviewer, and anki allows automated save every 10
minutes. Which means even if your device suddenly dies during a
streak, you only lose up to 10 minutes of review.

Still, this feature at least improve the case that was the most
dangerous for users (and frustrating for me)

I don't know the code of Reviewer2/ReviewerFragment. So this is only
enabled in the legacy reviewer. I hope the code is simple enough for
the feature to be easily ported to the other reviewers now that the
entire sync part of the code was decoupled from the DeckPicker.

Manually tested with:
* signin in (when no account present)
* full sync (requested locally and requested by ankiweb)
* normal sync
* changing the selected deck on the other device

To test it you should use a widget or a shortcut.

One point amy need to be changed. The little icon that state that
there is a not-uploaded change. While it's okay in the deckpicker, it
may not be ideal to have it as soon as we have the first review done.

* Fixes #17084
